Marlin/src/HAL/HAL_LPC1768/SoftwareSerial.h

```diff
@@ -71,12 +71,12 @@ class SoftwareSerial : public Stream
   static SoftwareSerial *active_object;
 
   // private methods
-  void recv() __attribute__((__always_inline__));
+  FORCE_INLINE void recv();
```

Comment:
I may be wrong, but I'm pretty sure `FORCE_INLINE` only works if the function body is defined in the same place. Since it's a compiler attribute and not a keyword like `inline` its behavior may differ. However, a function like this could still only inline in the same file where it is defined. All others still have to go through `static` linkage. Check the compiler results and see if the output is identical with and without these keywords.

Reply:
The original code was generating a "always_inline function might not be inlinable" warning due to the fact that the `always_inline` attribute was being set, but the function was not actually declared as inline. `FORCE_INLINE` is defined as below, which sets the attribute and declares the function as inline. 

`#define FORCE_INLINE __attribute__((always_inline)) inline`

Regarding the propose of the function being set as inline in the first place, perhaps @p3p can shed some light on that.

Reply:
Declaring a function `inline` without providing the function body is probably meaningless.